### PR TITLE
django-completion: update homepage and livecheck

### PR DESCRIPTION
**Pull request by chenrui333**
django-completion: update homepage and livecheck

### CODE REVIEW

**Review by samford — CHANGES_REQUESTED** [review]
The `stable` archive is from GitHub and the homepage doesn't link to archive files on GitHub, so I don't think it makes sense to check it.

We should simply update the `livecheck` block to use `url :stable` and continue to use the standard regex for Git tags (`/^v?(\d+(?:\.\d+)+)$/i`), as the check would be aligned with the `stable` source. The latest version from the Git tags is the same as the latest version on the homepage, so I don't see a functional need to check the homepage in this instance.

**Comment by chenrui333** [comment]
I thought we prefer the homepage over the github page?

**Comment by samford** [comment]
> I thought we prefer the homepage over the github page?

In general, we prefer to align the check with the stable source (GitHub here) and we tend to only diverge from that when there's a good reason. Under normal circumstances, it's fine to identify the latest release from the Git tags (or the "latest" release on GitHub, when necessary).

For example, if the formula uses an archive file from GitHub but identifying the latest release from GitHub isn't reliable (i.e., the Git tags have issues and the "latest" release is incorrect or unavailable), the first-party site can sometimes be a reliable indicator of the latest version if they link to GitHub archive files. This particular scenario doesn't happen very often but we encountered it recently, so that may be where you got this idea.

It's preferable to check the Git tags for this formula because `stable` is an archive from GitHub and the first-party site provides downloads for different archive files instead (i.e., not links to the GitHub archive files).

File: Formula/django-completion.rb [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+  homepage "https://www.djangoproject.com/"
   url "https://github.com/django/django/archive/3.1.5.tar.gz"
   sha256 "b481d9ffd70c51bd20039a3b5cd33ffe3ac188ba95d29b37e88e456472317eaf"
   license "BSD-3-Clause"
   head "https://github.com/django/django.git"
 
   livecheck do
-    url :head
-    regex(/^v?(\d+(?:\.\d+)+)$/i)
+    url :stable
```
Comment (samford) [inline Formula/django-completion.rb:10]:
```suggestion
    url :stable
    regex(/^v?(\d+(?:\.\d+)+)$/i)
```

We still need the existing regex.

**Review by samford — CHANGES_REQUESTED** [review]

**Review by samford — APPROVED** [review]

**Pull request merged**